pkg/model/azuremodel/api_loadbalancer.go

```diff
@@ -38,5 +45,120 @@ func (b *APILoadBalancerModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		return nil
 	}
 
-	return errors.New("using loadbalancer for API server is not yet implemented in Azure")
+	lbSpec := b.Cluster.Spec.API.LoadBalancer
+	if lbSpec == nil {
+		// Skipping API LB creation; not requested in Spec
+		return nil
+	}
+
+	switch lbSpec.Type {
+	case kops.LoadBalancerTypeInternal:
+		// OK
+	case kops.LoadBalancerTypePublic:
+		// TODO: Implement creating public ip and attach to public loadbalancer
+		return fmt.Errorf("only internal loadbalancer for API server is implemented in Azure")
+	default:
+		return fmt.Errorf("unhandled LoadBalancer type %q", lbSpec.Type)
+	}
+
+	// Create LoadBalancer for API ELB
+	lb := &azuretasks.LoadBalancer{
+		Name:          fi.String(b.NameForLoadBalancer()),
+		Lifecycle:     b.Lifecycle,
+		ResourceGroup: b.LinkToResourceGroup(),
+		Tags:          map[string]*string{},
+	}
+
+	switch lbSpec.Type {
+	case kops.LoadBalancerTypeInternal:
+		lb.RequirePublicIP = to.BoolPtr(false)
+		subnet, err := b.subnetForLoadBalancer()
+		if err != nil {
+			return err
+		}
+		lb.Subnet = b.LinkToAzureSubnet(subnet)
+	case kops.LoadBalancerTypePublic:
+		lb.RequirePublicIP = to.BoolPtr(true)
```

Comment:
Above we return an error that public isnt supported, but here we're setting values on the LB task as if it is supported. If it is supported we should drop the error message above and if it isn't supported then we dont need this case statement. I see some tests later in this PR that have `RequirePublicIP` set to true too.

Reply:
Public isn't supported because we would need to create a public ip address first and then attach it to the load balancer using its id. I can remove all the logic for creating a public load balancer if you think that would be better? i just thought it couldn't hurt having the logic there but stopping it from executing at the beginning.

Reply:
If you think we'll be able to add public support soon then I'm fine with keeping the underlying logic behind it in this PR rather than reverting and re-adding later. It sounds like we just need to create a "public ip address" azuretask and connect that to the load balancer task?

Reply:
yeah that's all that needs to be done for a public loadbalancer.